test/simulator/timestamp/src/session_simulator.cpp

```diff
@@ -59,3 +60,48 @@ session_simulator::commit_transaction()
 
     _txn_running = false;
 }
+
+void
+session_simulator::set_commit_timestamp(const uint64_t &ts)
+{
+    _commit_ts = ts;
+}
+
+void
+session_simulator::set_durable_timestamp(const uint64_t &ts)
+{
+    _durable_ts = ts;
+}
+
+void
+session_simulator::set_prepare_timestamp(const uint64_t &ts)
+{
+    _prepare_ts = ts;
+}
+
+void
+session_simulator::set_read_timestamp(const uint64_t &ts)
+{
+    _read_ts = ts;
+}
+
+int
+session_simulator::timestamp_transaction_uint(const std::string &ts_type, const uint64_t &ts)
+{
+    /* Zero timestamp is not permitted. */
+    if (ts == 0) {
+        WT_SIM_RET_MSG(EINVAL, "Illegal " + std::to_string(ts) + " timestamp: zero not permitted.");
+    }
+
+    if (ts_type == "commit") {
+        set_commit_timestamp(ts);
+    } else if (ts_type == "durable") {
+        set_durable_timestamp(ts);
+    } else if (ts_type == "prepare") {
+        set_prepare_timestamp(ts);
+    } else if (ts_type == "read") {
+        set_read_timestamp(ts);
+    }
```

Comment:
We should probably throw an error in an `else` statement.

Reply:
Yeah, that's a good idea.

Reply:
That relates to Vamsi's comment about an error earlier. WiredTiger just silently ignores if it doesn't have a correct timestamp type. @bvpvamsikrishna Should we add an error in this case?

Reply:
Yes, please add an error.

Reply:
Added.